### ISSUE TITLE
[0.54.4] Metro Bundler can't listen on port undefined

### ISSUE THREAD

**Issue by wouterds** [facebook/react-native#18673]
- [X] I have reviewed the [documentation](https://facebook.github.io/react-native)
- [X] I have searched [existing issues](https://github.com/facebook/react-native/issues)
- [X] I am using the [latest React Native version](https://github.com/facebook/react-native/releases)

Hi, I'm running the latest version of React Native and all goes well, except running from CLI to your device. Whenever I run `react-native run-ios --device` it fails with `Metro Bundler can't listen on port undefined`.

The issue has been fixed in https://github.com/facebook/react-native/pull/17983 however, but is scheduled for `0.55`. Which makes the current latest version just unusable and broken. Backport needed asap.

## Environment

```
Environment:
  OS: macOS High Sierra 10.13.3
  Node: 9.8.0
  Yarn: 1.3.2
  npm: 5.7.1
  Watchman: 4.9.0
  Xcode: Xcode 9.2 Build version 9C40b
  Android Studio: Not Found

Packages: (wanted => installed)
  react: ^16.3.0 => 16.3.0
  react-native: ^0.54.4 => 0.54.4
```

## Steps to Reproduce

Run `react-native run-ios --device` in any React Native project using the latest `0.54.4` release.

## Expected Behavior

Start the metro bundler correctly and continue compiling the application.

## Actual Behavior

Metro bundler fails because the port is undefined.

**Comment by jimji1005** [comment]
This is happening on 55 as well

`Environment:
  OS: macOS High Sierra 10.13.3
  Node: 8.6.0
  Yarn: 1.5.1
  npm: 5.6.0
  Watchman: 4.9.0
  Xcode: Xcode 9.3 Build version 9E145
  Android Studio: 3.1 AI-173.4670197

Packages: (wanted => installed)
  react: ^16.3.0-alpha.0 => 16.3.0
  react-native: 0.55.0 => 0.55.0`


❌  /Users/abc/Apps/testText/node_modules/react-native/React/Base/RCTBundleURLProvider.m:15:53: use of undeclared identifier 'undefined'; did you mean 'underline'?

const NSUInteger kRCTBundleURLProviderDefaultPort = RCT_METRO_PORT;

**Comment by hramos** [comment]
As this is already fixed in master, closing as Fixed.

0.55 should be getting released any moment. You may also discuss cherry-pick requests over at https://github.com/react-native-community/react-native-releases.

**Comment by ASteinheiser** [comment]
Switching to react-native: 0.52.2 solved this for me

**Comment by wouterds** [comment]
I see that the patch is included in `0.55`, but it does not work anymore (while it used to if you manually changed it in `0.54.4`). So we're still left with a broken build, afaics.

**Comment by hatlestad** [comment]
I've got the same issue as @jacktator and @wouterds
